inference_schema/parameter_types/pandas_parameter_type.py

```diff
@@ -101,11 +101,44 @@ def input_to_swagger(self):
         :rtype: dict
         """
         LIST_LIKE_ORIENTS = ('records', 'values')
-        json_sample = json.loads(self.sample_input.to_json(orient=self.orient))
+        json_sample = json.loads(self.sample_input.to_json(orient=self.orient, date_format='iso'))
 
         if self.orient in LIST_LIKE_ORIENTS:
             swagger_schema = get_swagger_for_list(json_sample)
         else:
             swagger_schema = get_swagger_for_nested_dict(json_sample)
 
+        if self.orient == 'records':
+            for column_name in self.sample_input.columns:
+                data_type = str(self.sample_input.dtypes[column_name])
+                if data_type.startswith('datetime'):
+                    swagger_schema['items']['properties'][str(column_name)]['format'] = 'date-time'
+                elif data_type.startswith('timedelta'):
+                    swagger_schema['items']['properties'][str(column_name)]['format'] = 'timedelta'
+        elif self.orient == 'index':
+            for row in swagger_schema['properties'].values():
+                for column_name in self.sample_input.columns:
+                    data_type = str(self.sample_input.dtypes[column_name])
+                    if data_type.startswith('datetime'):
+                        row['properties'][str(column_name)]['format'] = 'date-time'
+                    elif data_type.startswith('timedelta'):
```

Comment:
Is the `data_type` value in some complicated format? Using `startswith` seems a bit unusual for this sort of thing.

Reply:
Yeah, this is an example of one: 'datetime64[ns]'. I haven't been able to find a good document about all the possible options here, but from my understanding at least the part in the [] will change based on the time unit.

Reply:
So startswith is just the simple "catch all"